src/index.ts

```diff
@@ -108,8 +108,12 @@ export function newRidgeState<T>(iv: T, o?: Options<T>): StateWithValue<T> {
     se: (state: T) => TSelected,
     eq = equ as any
   ): TSelected {
-    const [rv] = use();
-    return me(se(rv), eq);
+    const [l, s] = R.useState<TSelected>();
+    sub((nv) => {
+      const v = se(nv);
+      if (!eq(v, l)) s(v);
+    });
+    return l;
```

Comment:
Looks like it's not initialized properly. Recommend updating this `useState(() => selector(v));` and making a new PR since the code has divergged.

```js
  function useSelector(selector, comparator = equ) {
    const [state, setState] = useState(() => selector(v));
    subscribe(() => {
      const selectedValue = selector(v);
      if (!comparator(selectedValue, state)) setState(selectedValue);
    });
    return state;
  }
```